pkg/providers/azure.go

```diff
@@ -26,6 +26,7 @@ type SetUpCmd struct {
 	tenantId          string
 	appObjectId       string
 	spObjectId        string
+	AzTenantClient    azTenantClient
```

Comment:
if were going to have many interfaces associated with the az client, we should pull the clients into a single struct and have it be initialized.

Have an AzClient struct that will contain the sub tenantClient, appsClient, etc interfaces.